src/Functions/castOrDefault.cpp

```diff
@@ -376,66 +376,475 @@ SELECT accurateCastOrDefault('abc', 'UInt32')
 
     factory.registerFunction("toDecimal32OrDefault", [](ContextPtr context)
-        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal32OrDefault", createDecimalMaxPrecision<Decimal32>(0)); });
+        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal32OrDefault", createDecimalMaxPrecision<Decimal32>(0)); },
+        toDecimal32OrDefault_documentation);
+
+    FunctionDocumentation::Description toDecimal64OrDefault_description = R"(
+Like [`toDecimal64`](#todecimal64), this function converts an input value to a value of type [Decimal(18, S)](../data-types/decimal.md) but returns the default value in case of an error.
+    )";
+    FunctionDocumentation::Syntax toDecimal64OrDefault_syntax = "toDecimal64OrDefault(expr, S[, default])";
+    FunctionDocumentation::Arguments toDecimal64OrDefault_arguments = {
+        {"expr", "A String representation of a number.", {"String"}},
+        {"S", "Scale parameter between 0 and 18, specifying how many digits the fractional part of a number can have.", {"UInt8"}},
+        {"default", "Optional. The default value to return if parsing to type Decimal64(S) is unsuccessful.", {"Decimal64(S)"}}
+    };
+    FunctionDocumentation::ReturnedValue toDecimal64OrDefault_returned_value = {"Value of type Decimal(18, S) if successful, otherwise returns the default value if passed or 0 if not.", {"Decimal64(S)"}};
+    FunctionDocumentation::Examples toDecimal64OrDefault_examples = {
+        {"Successful conversion", "SELECT toDecimal64OrDefault(toString(0.0001), 18)", "0.0001"},
+        {"Failed conversion", "SELECT toDecimal64OrDefault('Inf', 0, CAST('-1', 'Decimal64(0)'))", "-1"}
+    };
+    FunctionDocumentation::Category toDecimal64OrDefault_category = FunctionDocumentation::Category::TypeConversion;
+    FunctionDocumentation::IntroducedIn toDecimal64OrDefault_introduced_in = {21, 11};
+    FunctionDocumentation toDecimal64OrDefault_documentation = {toDecimal64OrDefault_description, toDecimal64OrDefault_syntax, toDecimal64OrDefault_arguments, toDecimal64OrDefault_returned_value, toDecimal64OrDefault_examples, toDecimal64OrDefault_introduced_in, toDecimal64OrDefault_category};
+
     factory.registerFunction("toDecimal64OrDefault", [](ContextPtr context)
-        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal64OrDefault", createDecimalMaxPrecision<Decimal64>(0)); });
+        { return std::make_shared<FunctionCastOrDefaultTyped>(context, "toDecimal64OrDefault", createDecimalMaxPrecision<Decimal64>(0)); },
+        toDecimal64OrDefault_documentation);
+
+    FunctionDocumentation::Description toDecimal128OrDefault_description = R"(
+Like [`toDecimal128`](#todecimal128), this function converts an input value to a value of type [Decimal(38, S)](../data-types/decimal.md) but returns the default value in case of an error.
```

Comment:
```suggestion
Like [`toDecimal128`](#toDecimal128), this function converts an input value to a value of type [Decimal(38, S)](../data-types/decimal.md) but returns the default value in case of an error.
```